### PR TITLE
test ocr_telemetry_feed_message_report_req_median metric

### PR DESCRIPTION
**Pull request by gheorghestrimtu**
Added a test for the `ocr_telemetry_feed_message_report_req_median` metric.
Modified the `OCRSetupInputs` in `testcommon.ocr.go` to use `mockserver` for the adapter. Also, now every chainlink can be given different values to read from the "adapter", which allows for testing more scenarios.

### CODE REVIEW

**Review by tateexon — APPROVED** [review]

File: environment/environment_templates.go [hunk 1]
```diff
@@ -315,7 +316,7 @@ func NewChainlinkClusterForAlertsTesting(nodeCount int) K8sEnvSpecInit {
 	}
 
 	dependencyGroup := getBasicDependencyGroup()
-	addServicesForTestingAlertsToDependencyGroup(dependencyGroup, nodeCount)
+	addServicesForTestingObservabilityToDependencyGroup(dependencyGroup, nodeCount)
```
Comment (skudasov) [inline environment/environment_templates.go:319]: Do we really need a method with such a long name that only pushes to a slice? Maybe slice is enough I think

File: suite/observability/metrics_test.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+		By("Kicking off first round")
+
+		testcommon.StartNewRound(testSetup, 1)
+
+		By("Comparing the metrics value with the expected value")
+
+		prometheus, err := environment.GetPrometheusClientFromEnv(testSetup.SuiteSetup.Env)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		Eventually(func(g Gomega) int {
```
Comment (skudasov) [inline suite/observability/metrics_test.go:46]: Better to use `retry` for consistency so we don't have two libs for one purpose

**Review by skudasov — APPROVED** [review]

**Review by skudasov — CHANGES_REQUESTED** [review]
`make test_ocr` panics, seems it need new mock server in setup

File: environment/environment_templates.go (hunk 1 above)
Comment (gheorghestrimtu) [reply environment/environment_templates.go]: removed this function

File: suite/observability/metrics_test.go (hunk 2 above)
Comment (gheorghestrimtu) [reply suite/observability/metrics_test.go:46]: changed all retry's in tests to eventually

**Review by skudasov — APPROVED** [review]

**Pull request merged**